### PR TITLE
Add Choice step.

### PR DESCRIPTION
**Pull request by gtopper**

### CODE REVIEW

File: tests/test_flow.py [hunk 1]
```diff
@@ -259,3 +259,27 @@ async def async_test_error_async_flow():
 def test_error_async_flow():
     loop = asyncio.new_event_loop()
     loop.run_until_complete(async_test_error_async_flow())
+
+
+def test_choice():
+    small_reduce = build_flow([
```
Comment (talIguaz) [inline tests/test_flow.py]: why isn't it just `small_reduce= Reduce(0, lambda acc, x: acc + x)`

File: storey/flow.py [hunk 2]
```diff
@@ -56,6 +56,32 @@ async def _do_downstream(self, event):
...
+            return await super()._do_downstream(event)
+        chosen_outlet = None
+        for outlet, condition in self._choice_array:
+            if condition(event.element):
+                chosen_outlet = outlet
+                break
+        if chosen_outlet:
+            await chosen_outlet._do(event)
+        elif self._default:
+            await self._default._do(event)
```
Comment (talIguaz) [inline storey/flow.py:82]: `default` is None be default, so you can get an error here

Comment (gtopper) [reply storey/flow.py:82]: Hence elif.

**Review by talIguaz — APPROVED** [review]
LGTM

**Pull request merged**